index.rst

```diff
@@ -2789,63 +2789,94 @@ rows, rather than some finite number (thousands or less).
-first query for the current chunk id, the waiting query is dispatched.
-Queries are dispatched from the front of the queue until there are no
-more slots or matching queries. If the active queue is empty, the active
-and pending queues are swapped. If the waiting query's chunk id is not
-the current chunk id, it may not be dispatched until (a) at least one
-query for the current chunk id has been completed, signifying that the
-chunk is likely to be completely cached and (b) no queries for other
-chunk id are in-flight. If (a) is not true, the current chunk id is
-likely still being read off disk and a query referencing a different
-chunk will compete for I/O and reduce total disk bandwidth. If (b) is
-not true, then there are other queries that rely on past cached chunks,
-and a new chunk id dispatch would likely kick out those past chunks from
-the cache and force them to incur their own I/O costs.
-
-The scan scheduler utilizes distinct pairs of queues for each sequential
-resource. It is thus able to track and support multiple scans
-simultaneously, provided only one scan is in progress for each disk
-(sequential data source). This maximizes each disk's I/O bandwidth.
-Whereas the I/O resources are treated as parallel and independent, the
-CPU cores on workers are shared, and the scheduler seeks to strike a
-balance between maximizing I/O bandwidth and maximizing scan reuse.
-Maximizing I/O bandwidth would always seek to start new chunk scans
-whenever a disk is idle, skipping over queries that can execute on
-cached data. This makes scans complete faster, reducing scan latency,
-but reduces the amount of useful work done per-scan. On the other hand,
-maximizing scan reuse would always select queries that operated on
-already-scanned (and thus cached) chunks, maximizing CPU efficiency
-because it minimizes the working set of data and thus the pressure on
-the system cache and memory subsystem. This may leave some disks idle,
-but is likely to produce the maximum throughput, unless the skew in CPU
-costs among scanning queries is too great.
+to implement shared scans.  Instead, it issues query fragments ordered to
+maximize locality of access in data and time, and tries to lock the files
+associated with the tables in memory as much as possible. Using the
+identified scan tables and their ratings, the worker places them on the
+appropriate scheduler. There will be at least three schedulers. One for
+queries expected to be complete in under an hour, which are expected to
+be related to the Object table. One for queries expected to take less than
+eight hours, expected to be related to Object_Extra. And one for scans
+expected to take twelve hours for ForcedSource and/or Source tables. The
+reasoning being that a single slow query can impede the progress of a
+shared scan and all the other user queries on that scan. There may be a
+need for another scheduler to handle queries taking more than 12 hours.
```

Comment:
Oh, I see now. However, then it should be cleared in my previous comment that the third scheduler is for queries between 8 and 12 hours.